src/core/EventEmitter.ts

```diff
@@ -16,31 +16,59 @@ type EventCallback<T extends EventMap, K extends EventKey<T>>
       : (event: T[K]) => any;
 
 class EventEmitter<T extends EventMap> {
-  private _listenerMap: Map<EventKey<T>, Array<EventCallback<T, EventKey<T>>>>;
+  private _listenerMap: {
```

Comment:
Why didn't you use Component?

Reply:
I have to change that later, but not in this PR as it's just adding some more tests.
Adding Component as a dependency would require a version update as it changes some event parameters.
That code change is just a copy of what I'm using on my private repository, with a slightly more improvement.

Reply:
OK. After writing TC, it would be better to make it a task and convert it to a component.